File: doc/docs/examples/00_dataloading.scala
import edu.ucr.cs.bdlab.beast._
import edu.ucr.cs.bdlab.beast.common.{BeastOptions, CLIOperation}
import org.apache.hadoop.fs.{Path, PathFilter}
import org.apache.spark.{SparkConf, SparkContext}
import org.apache.spark.rdd.RDD
import org.apache.spark.sql.SparkSession

object dataloading {

def run(sc: SparkContext): RasterRDD[Int] = {
    val path =
      "/Users/clockorangezoe/Documents/phd_projects/code/geoAI/RDProLLMagent/data/landsat8/LA/B4/LC08_L2SP_040037_20250827_20250903_02_T1_SR_B4.TIF"

    val raster: RasterRDD[Int] = sc.geoTiff[Int](path)

    // Force execution + small driver-side sanity check
    val t = raster.first
    val v = t.getPixelValue(t.x1, t.y1)
    println(s"[dataloading] pixelType=${t.pixelType}, sample=$v, partitions=${raster.getNumPartitions}")

    raster
  }
}

File: doc/docs/examples/00_dataloading.py
from osgeo import gdal

def run():
    path = (
        "/Users/clockorangezoe/Documents/phd_projects/code/geoAI/RDProLLMagent/"
        "data/landsat8/LA/B4/LC08_L2SP_040037_20250827_20250903_02_T1_SR_B4.TIF"
    )

    ds = gdal.Open(path, gdal.GA_ReadOnly)
    if ds is None:
        raise FileNotFoundError(f"GDAL could not open: {path}")

    band = ds.GetRasterBand(1)

    # Pixel type (similar idea to RDPro pixelType)
    dtype_name = gdal.GetDataTypeName(band.DataType)

    # Force execution: read one pixel (top-left of raster)
    sample = band.ReadAsArray(0, 0, 1, 1)[0, 0]

    # Optional: raster size / "partitions"-ish info (GDAL doesn't partition like Spark)
    width, height = ds.RasterXSize, ds.RasterYSize

    print(
        f"[dataloading] pixelType={dtype_name}, sample={sample}, "
        f"size={width}x{height}, bands={ds.RasterCount}"
    )

    return ds


if __name__ == "__main__":
    run()
